File: Gestion A C/models/ArticleVente.php
<?php
require_once 'Article.php';
require_once 'Production.php';
require_once 'Vente.php';

class ArticleVente extends Article {
    private static int $nbAv = 0;
    private static array $productions = [];
    private static array $ventes = [];

    // Constructeur avec ou sans paramètres
    public function __construct(string $nomArticle = "", float $prix = 0, int $quantite = 0) {
        parent::__construct();
        if ($nomArticle !== "" && $prix !== 0 && $quantite !== 0) {
            self::$nbAv++;
            $this->id = "AV-" . self::$nbAv;
            $this->nomArticle = $nomArticle;
            $this->prix = $prix;
            $this->quantiteStock = $quantite;
            $this->montantStock = $prix * $quantite;
            $this->type = "Article Vente";
        }
    }

    // Liste des productions
    public static function getProductions(): array {
        return self::$productions;
    }

    public static function addProduction(Production $production): void {
        self::$productions[] = $production;
    }

    // Liste des ventes
    public static function getVentes(): array {
        return self::$ventes;
    }

    public static function addVente(Vente $vente): void {
        self::$ventes[] = $vente;
    }
}

File: Gestion Atelier De Couture/src/entity/ArticleVente.java
package entity;

import java.util.ArrayList;

public class ArticleVente extends Article {

    private static int nbAv;
    private static ArrayList<Production> productions = new ArrayList<>();
    private static ArrayList<Vente> ventes = new ArrayList<>();

    public ArticleVente(String nomArticle, double prix, int quantite) {
        super();
        nbAv++;
        this.id = "AV-" + nbAv;
        this.nomArticle = nomArticle;
        this.prix = prix;
        this.quantiteStock = quantite;
        this.montantStock = prix * quantite;
        this.type = "Article Vente";
    }
    
    public ArticleVente() {
        super();
    }

    public static ArrayList<Production> getProductions() {
        return productions;
    }

    public static void addProduction(Production production) {
        productions.add(production);
    }

    public static ArrayList<Vente> getVentes() {
        return ventes;
    }

    public static void addVente(Vente vente) {
        ventes.add(vente);
    }

}
